client-v2/src/main/java/com/clickhouse/client/api/data_formats/internal/AbstractBinaryFormatReader.java

```diff
@@ -737,4 +732,19 @@ public ClickHouseBitmap getClickHouseBitmap(int index) {
     public void close() throws Exception {
         input.close();
     }
+
+    @Override
+    public int getColumnCount() {
+        return currentRecord.size();

```

Comment:
It should be returned from schema because it may be a virtual columns (for example, when we do so for tuples)